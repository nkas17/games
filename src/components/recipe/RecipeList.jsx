import React from 'react';
import PropTypes from 'prop-types';
import RecipeListRow from './RecipeListRow'

/**
 * List of recipes
 */
const RecipeList = ({recipes}) => {
	return (
		<table className="table">
			<thead>
				<tr>
					<th>&nbsp;</th>
					<th>title</th>
					<th>description</th>
				</tr>
			</thead>
			<tbody>
				{recipes.map(recipe =>
				(<RecipeListRow 
					key={recipe.id} 
					recipe={recipe}
				/>)
				)}
			</tbody>
		</table>	
	)
}

RecipeList.propTypes = {
	/**
	 * list of recipes to display
	 */
	recipes: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default RecipeList;
