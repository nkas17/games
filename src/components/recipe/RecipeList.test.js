import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import RecipeList from './RecipeList';

/* eslint-disable no-undef */

const setup = () => (
	{
		recipes: [
			{
				id: 'tacos',
				title: 'tacos',
				category: 'other',
				description: 'Tasty homemade tacos - hard or soft',
				ingredients: [
					{
						id: 'cheese',
						quantity: '8',
						unit: 'oz'
					},
					{
						id: 'ground beef',
						quantity: '1.5',
						unit: 'lbs'
					},
					{
						id: 'tortilla',
						quantity: '1',
						unit: ''
					}
				]
			},
			{
				id: 'quiche',
				title: 'quiche',
				category: 'other',
				description: 'Delicious Spinach Quiche and Ham & Cauliflower quiche',
				ingredients: [
					{
						id: 'cheese',
						quantity: '8',
						unit: 'oz'
					},
					{
						id: 'pie-crust',
						quantity: '2',
						unit: ''
					},
					{
						id: 'egg',
						quantity: '7',
						unit: ''
					}
				]
			}
		]
	}
);

const renderWithRouter = node => renderer.create(<Router>{node}</Router>);

describe('RecipeList', () => {
	it('renders correctly', () => {
		const tree = renderWithRouter(
			<RecipeList {...setup()} />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
