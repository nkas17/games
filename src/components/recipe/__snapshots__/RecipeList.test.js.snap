// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeList renders correctly 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
         
      </th>
      <th>
        title
      </th>
      <th>
        description
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
         
      </td>
      <td>
        <a
          href="recipe/tacos"
          onClick={[Function]}
        >
          tacos
        </a>
      </td>
      <td>
        Tasty homemade tacos - hard or soft
      </td>
    </tr>
    <tr>
      <td>
         
      </td>
      <td>
        <a
          href="recipe/quiche"
          onClick={[Function]}
        >
          quiche
        </a>
      </td>
      <td>
        Delicious Spinach Quiche and Ham & Cauliflower quiche
      </td>
    </tr>
  </tbody>
</table>
`;
